### PR TITLE
transaction control and queries

### PR DESCRIPTION
**Pull request by J2ZROMERO**
- Add transactions control.
- Add data rows.
- Update rows.
- Add species column.
[Word document content of images of queries](https://github.com/J2ZROMERO/Vet-Clinic/files/10233641/queries.docx)

### CODE REVIEW

**Review by rloterh — COMMENTED** [review]
Hi Team, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, for which the Pull Request description does not contain the appropriate images attached. Kindly ensure to attach the screenshots of your query and its results.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by J2ZROMERO** [comment]
- Add transactions control.
- Add data rows.
- Update rows.
- Add species column.

2022-12-14_22h51_45
2022-12-14_22h52_01
2022-12-14_22h52_13
2022-12-14_22h52_34
2022-12-14_22h52_43
2022-12-14_22h52_50
2022-12-14_22h52_59
2022-12-14_22h53_10

File: data.sql [hunk 1]
```diff
@@ -1,6 +1,9 @@
 /* Populate database with sample data. */
 
-insert into animals values (1,'Agumon','2020-02-03',0,false,10.23);
-insert into animals values (2,'Gabumon','2018-11-15',2,true,8);
-insert into animals values (3,'Pikachu','2021-01-07',1,true,15.04);
-insert into animals values (4,'Devimon','2017-05-12',5,true,11);
+insert into animals(id,name,date_of_birth,escape_attempts,neutered,weigth_kg) values (5,'Charmander','2020-02-08',0,false,-11);
+insert into animals(id,name,date_of_birth,escape_attempts,neutered,weigth_kg) values (6,'Plantmon','2021-11-15',2,true,-5.7);
+insert into animals(id,name,date_of_birth,escape_attempts,neutered,weigth_kg) values (7,'Squirtle','1993-03-02',3,false,-12.13);
```
Comment (Davidosky007) [inline data.sql]: * Kindly add the correct date of birth for the animal named `Squirtle` DOB: `He was born on Apr 2nd, 1993,` [According to the requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=He%20was%20born%20on%20Apr%202nd%2C%201993%2C)

File: queries.sql [hunk 2]
```diff
@@ -1,11 +1,45 @@
...
-SELECT * from animals WHERE name like '%mon';
-select name from animals where date_of_birth between '2016-01-01' and '2019-12-31';
-select name from animals where neutered = true and escape_attempts < 3;
-select date_of_birth from animals where name in ('Agumon','Pikachu');
-select name,escape_attempts from animals where weigth_kg > 10.5;
-select * from animals where neutered = true;
-select * from animals where name not  in ('Gabumon');
-select * from animals where weigth_kg between 10.4 and 17.3;
+begin;
+ALTER TABLE animals RENAME COLUMN species TO unspecified;
```
Comment (Davidosky007) [inline queries.sql:10]: * Kindly verify that the change was made after `species = unspecified` and verify that the species columns went back to the state before the transaction.
 image

File: queries.sql [hunk 3]
```diff
@@ -1,11 +1,45 @@
...
+rollback;
+
+begin;
+update animals set species = 'digimon' where name like '%mon';
+update animals set species = 'pokemon' where name not like '%mon';
```
Comment (Davidosky007) [inline queries.sql:15]: * Kindly verify that the change was made  here as requested in the[ project requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Verify%20that%20change%20was%20made%20and%20persists%20after%20commit.)

File: queries.sql [hunk 4]
```diff
@@ -1,11 +1,45 @@
...
+commit;
```
Comment (Davidosky007) [inline queries.sql:16]: * Kindly `verify that change persists after commit`

File: queries.sql [hunk 5]
```diff
@@ -1,11 +1,45 @@
...
+update animals set weigth_kg =  (weigth_kg * -1) where weigth_kg < 0 ;
+commit;
+
+
+
+
+
+
+select count(id) from animals;
+select count(id) from animals where escape_attempts > 0;
```
Comment (Davidosky007) [inline queries.sql]: * Kindly adjust the query for this transaction `How many animals have never tried to escape?` escape_attempts should be = 0 or <= 0 not greater than 0

File: queries.sql [hunk 6]
```diff
@@ -1,11 +1,45 @@
...
+select avg(weigth_kg) from animals ;
```
Comment (Davidosky007) [inline queries.sql:40]: * Kindly add a query for this transaction `Who escapes the most, neutered or not neutered animals?` as its missing

File: queries.sql [hunk 7]
```diff
@@ -1,11 +1,45 @@
...
+select name from animals where weigth_kg = ( select max(weigth_kg) from animals);
```
Comment (Davidosky007) [inline queries.sql]: * Kindly add the correct query for this transaction `What is the minimum and maximum weight of each type of animal?` as your current query doesn't answer the question( What is the minimum and maximum weight of each type of animal?)

File: queries.sql [hunk 8]
```diff
@@ -1,11 +1,45 @@
...
+select name, weigth_kg from animals;
+select name,avg(escape_attempts) from animals where date_of_birth between '1990-01-01' and  '2000-12-31' group by name;
```
Comment (Davidosky007) [inline queries.sql]: * Kindly write your query efficiently by selecting your query by animal type ( species ) as the question says ( What is the average number of escape attempts **per animal type** of those born between 1990 and 2000?)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @J2ZROMERO 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* Good looking readme file 👍 
* Well written PR description 👍 

## Required Changes ♻️
* **I noticed that the queries you have in the screenshots on your PR are different from the ones you wrote, please ensure that the results you show on your PR and queries are the same results you get when you run your queries 🙏 this would ensure proper mastery of this milestone project** 
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by J2ZROMERO** [comment]
- Fix queries

2022-12-15_09h19_20

2022-12-15_09h14_50
2022-12-15_09h23_28
2022-12-15_09h25_54

2022-12-15_09h31_19
2022-12-15_09h34_13
2022-12-15_09h36_19

**Review by DeliceLydia — APPROVED** [review]
## Hi @J2ZROMERO, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**